src/vs/server/node/webClientServer.ts

```diff
@@ -128,6 +139,38 @@ export class WebClientServer {
 		return serveFile(this._logService, req, res, filePath, headers);
 	}
 
+	/**
+	 * Handle extension resources
+	 */
+	private async _handleWebExtensionResource(req: http.IncomingMessage, res: http.ServerResponse, parsedUrl: url.UrlWithParsedQuery): Promise<void> {
+		// Strip `/web-extension-resource/` from the path
+		const normalizedPathname = decodeURIComponent(parsedUrl.pathname!); // support paths that are uri-encoded (e.g. spaces => %20)
+		const path = normalize(normalizedPathname.substr('/web-extension-resource/'.length));
+
+		const url = URI.parse(path).with({
+			scheme: this._productService.extensionsGallery?.resourceUrlTemplate ? URI.parse(this._productService.extensionsGallery.resourceUrlTemplate).scheme : 'https',
+			authority: path.substring(0, path.indexOf('/')),
+			path: path.substring(path.indexOf('/') + 1)
+		}).toString(true);
+
+		const headers: IHeaders = {};
+		for (const header of req.rawHeaders) {
+			if (req.headers[header]) {
+				headers[header] = req.headers[header]![0];
```

Comment:
copying all headers seems risky. This will include potentially sensitive headers such as `Cookie`.

Reply:
😱

Reply:
Changed it to send selective headers